File: OTEAServer/Models/SubSubAmbit.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace OTEAServer.Models
{
    /// <summary>
    /// Model class for second level division of ambits
    /// Author: Pablo Ahíta del Barrio
    /// Version: 1
    /// </summary>
    public class SubSubAmbit
    {
        /// <summary>
        /// Class constructor
        /// </summary>
        /// <param name="idSubSubAmbit">Second division of ambit identifier</param>
        /// <param name="idSubAmbit">First division of ambit identifier</param>
        /// <param name="idAmbit">Ambit identifier</param>
        /// <param name="descriptionEnglish">SubSubAmbit description in English</param>
        /// <param name="descriptionSpanish">SubSubAmbit description in Spanish</param>
        /// <param name="descriptionFrench">SubSubAmbit description in French</param>
        /// <param name="descriptionBasque">SubSubAmbit description in Basque</param>
        /// <param name="descriptionCatalan">SubSubAmbit description in Catalan</param>
        /// <param name="descriptionDutch">SubSubAmbit description in Dutch</param>
        /// <param name="descriptionGalician">SubSubAmbit description in Galician</param>
        /// <param name="descriptionGerman">SubSubAmbit description in German</param>
        /// <param name="descriptionItalian">SubSubAmbit description in Italian</param>
        /// <param name="descriptionPortuguese">SubSubAmbit description in Portuguese</param>
        public SubSubAmbit(int idSubSubAmbit, int idSubAmbit, int idAmbit, String descriptionEnglish, String descriptionSpanish, string descriptionFrench, string descriptionBasque, string descriptionCatalan, string descriptionDutch, string descriptionGalician, string descriptionGerman, string descriptionItalian, string descriptionPortuguese)
        {
            this.idSubSubAmbit = idSubSubAmbit;
            this.idSubAmbit = idSubAmbit;
            this.idAmbit = idAmbit;
            this.descriptionSpanish = descriptionSpanish;
            this.descriptionEnglish = descriptionEnglish;
            this.descriptionFrench = descriptionFrench;
            this.descriptionBasque = descriptionBasque;
            this.descriptionCatalan = descriptionCatalan;
            this.descriptionDutch = descriptionDutch;
            this.descriptionGalician = descriptionGalician;
            this.descriptionGerman = descriptionGerman;
            this.descriptionItalian = descriptionItalian;
            this.descriptionPortuguese = descriptionPortuguese;
        }

        /// <summary>
        /// Second division of ambit identifier
        /// </summary>
        [JsonPropertyName("idSubSubAmbit")]
        public int idSubSubAmbit { get; set; }

        /// <summary>
        /// First division of ambit identifier
        /// </summary>
        [JsonPropertyName("idSubAmbit")]
        public int idSubAmbit { get; set; }

        /// <summary>
        /// Ambit identifier
        /// </summary>
        [Key]
        [JsonPropertyName("idAmbit")]
        public int idAmbit { get; set; }

        /// <summary>
        /// SubSubAmbit description in Spanish
        /// </summary>
        [JsonPropertyName("descriptionSpanish")]
        public string descriptionSpanish { get; set; }

        /// <summary>
        /// SubSubAmbit description in English
        /// </summary>
        [JsonPropertyName("descriptionEnglish")]
        public string descriptionEnglish { get; set; }

        /// <summary>
        /// SubSubAmbit description in French
        /// </summary>
        [JsonPropertyName("descriptionFrench")]
        public string descriptionFrench { get; set; }

        /// <summary>
        /// SubSubAmbit description in Basque
        /// </summary>
        [JsonPropertyName("descriptionBasque")]
        public string descriptionBasque { get; set; }

        /// <summary>
        /// SubSubAmbit description in Catalan
        /// </summary>
        [JsonPropertyName("descriptionCatalan")]
        public string descriptionCatalan { get; set; }

        /// <summary>
        /// SubSubAmbit description in Dutch
        /// </summary>
        [JsonPropertyName("descriptionDutch")]
        public string descriptionDutch { get; set; }


        /// <summary>
        /// SubSubAmbit description in Galician
        /// </summary>
        [JsonPropertyName("descriptionGalician")]
        public string descriptionGalician { get; set; }

        /// <summary>
        /// SubSubAmbit description in German
        /// </summary>
        [JsonPropertyName("descriptionGerman")]
        public string descriptionGerman { get; set; }

        /// <summary>
        /// SubSubAmbit description in Italian
        /// </summary>
        [JsonPropertyName("descriptionItalian")]
        public string descriptionItalian { get; set; }

        /// <summary>
        /// SubSubAmbit description in Portuguese
        /// </summary>
        [JsonPropertyName("descriptionPortuguese")]
        public string descriptionPortuguese { get; set; }
    }
}

File: connection/src/main/java/cli/indicators/SubSubAmbit.java
package cli.indicators;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

/**
 * Model class for second level division of ambits
 *
 * @author Pablo Ahíta del Barrio
 * @version 1
 * */
public class SubSubAmbit implements Serializable {
    
    /**Second level division of ambit identifier*/
    @SerializedName("idSubSubAmbit")
    public int idSubSubAmbit;

    /**First level division of ambit identifier*/
    @SerializedName("idSubAmbit")
    public int idSubAmbit;
    
    /**Ambit identifier*/
    @SerializedName("idAmbit")
    public int idAmbit;

    /**SubSubAmbit description in English*/
    @SerializedName("descriptionEnglish")
    public String descriptionEnglish;

    /**SubSubAmbit description in Spanish*/
    @SerializedName("descriptionSpanish")
    public String descriptionSpanish;

    /**SubSubAmbit description in French*/
    @SerializedName("descriptionFrench")
    public String descriptionFrench;

    /**SubSubAmbit description in Basque*/
    @SerializedName("descriptionBasque")
    public String descriptionBasque;

    /**SubSubAmbit description in Catalan*/
    @SerializedName("descriptionCatalan")
    public String descriptionCatalan;

    /**SubSubAmbit description in Dutch*/
    @SerializedName("descriptionDutch")
    public String descriptionDutch;

    /**SubSubAmbit description in Galician*/
    @SerializedName("descriptionGalician")
    public String descriptionGalician;

    /**SubSubAmbit description in German*/
    @SerializedName("descriptionGerman")
    public String descriptionGerman;

    /**SubSubAmbit description in Italian*/
    @SerializedName("descriptionItalian")
    public String descriptionItalian;

    /**SubSubAmbit description in Portuguese*/
    @SerializedName("descriptionPortuguese")
    public String descriptionPortuguese;

    /**
     * Class constructor
     *
     * @param idSubSubAmbit - SubSubAmbit identifier
     * @param idSubAmbit - SubAmbit identifier
     * @param idAmbit - Ambit identifier
     * @param descriptionEnglish - SubSubAmbit description in English
     * @param descriptionSpanish - SubSubAmbit description in Spanish
     * @param descriptionFrench - SubSubAmbit description in French
     * @param descriptionBasque - SubSubAmbit description in Basque
     * @param descriptionCatalan - SubSubAmbit description in Catalan
     * @param descriptionDutch - SubSubAmbit description in Dutch
     * @param descriptionGalician - SubSubAmbit description in Galician
     * @param descriptionGerman - SubSubAmbit description in German
     * @param descriptionItalian - SubSubAmbit description in Italian
     * @param descriptionPortuguese - SubSubAmbit description in Portuguese
     * */
    public SubSubAmbit(int idSubSubAmbit, int idSubAmbit, int idAmbit, String descriptionEnglish, String descriptionSpanish, String descriptionFrench,String descriptionBasque, String descriptionCatalan, String descriptionDutch,String descriptionGalician, String descriptionGerman, String descriptionItalian,String descriptionPortuguese){
        setIdSubSubAmbit(idSubSubAmbit);
        setIdSubAmbit(idSubAmbit);
        setIdAmbit(idAmbit);
        setDescriptionEnglish(descriptionEnglish);
        setDescriptionSpanish(descriptionSpanish);
        setDescriptionFrench(descriptionFrench);
        setDescriptionBasque(descriptionBasque);
        setDescriptionCatalan(descriptionCatalan);
        setDescriptionDutch(descriptionDutch);
        setDescriptionGalician(descriptionGalician);
        setDescriptionGerman(descriptionGerman);
        setDescriptionItalian(descriptionItalian);
        setDescriptionPortuguese(descriptionPortuguese);
    }

    /**
     * Method that obtains the subSubAmbit identifier
     *
     * @return SubSubAmbit identifier
     * */
    public int getIdSubSubAmbit() {
        return idSubSubAmbit;
    }

    /**
     * Method that sets the new subSubAmbit identifier
     *
     * @param idSubSubAmbit - SubSubAmbit identifier
     * */
    public void setIdSubSubAmbit(int idSubSubAmbit) {
        this.idSubSubAmbit = idSubSubAmbit;
    }

    /**
     * Method that obtains the subAmbit identifier
     *
     * @return SubAmbit identifier
     * */
    public int getIdSubAmbit() {
        return idSubAmbit;
    }

    /**
     * Method that sets the new subAmbit identifier
     *
     * @param idSubAmbit - SubAmbit identifier
     * */
    public void setIdSubAmbit(int idSubAmbit) {
        this.idSubAmbit = idSubAmbit;
    }

    /**
     * Method that obtains the ambit identifier
     *
     * @return Ambit identifier
     * */
    public int getIdAmbit() {
        return idAmbit;
    }

    /**
     * Method that sets the new ambit identifier
     *
     * @param idAmbit - Ambit identifier
     * */
    public void setIdAmbit(int idAmbit) {
        this.idAmbit = idAmbit;
    }

    /**
     * Method that obtains the subSubAmbit description in English
     *
     * @return Ambit description in English
     * */
    public String getDescriptionEnglish() {
        return descriptionEnglish;
    }

    /**
     * Method that sets the new subSubAmbit description in English
     *
     * @param descriptionEnglish - Ambit description in English
     * */
    public void setDescriptionEnglish(String descriptionEnglish) {
        this.descriptionEnglish = descriptionEnglish;
    }

    /**
     * Method that obtains the subSubAmbit description in Spanish
     *
     * @return Ambit description in Spanish
     * */
    public String getDescriptionSpanish() {
        return descriptionSpanish;
    }

    /**
     * Method that sets the new subSubAmbit description in Spanish
     *
     * @param descriptionSpanish - Ambit description in Spanish
     * */
    public void setDescriptionSpanish(String descriptionSpanish) {
        this.descriptionSpanish = descriptionSpanish;
    }

    /**
     * Method that obtains the subSubAmbit description in French
     *
     * @return Ambit description in French
     * */
    public String getDescriptionFrench() {
        return descriptionFrench;
    }

    /**
     * Method that sets the new subSubAmbit description in Portuguese
     *
     * @param descriptionFrench - Ambit description in French
     * */
    public void setDescriptionFrench(String descriptionFrench) {
        this.descriptionFrench = descriptionFrench;
    }

    /**
     * Method that obtains the subSubAmbit description in Basque
     *
     * @return Ambit description in Basque
     * */
    public String getDescriptionBasque() {
        return descriptionBasque;
    }

    /**
     * Method that sets the new subSubAmbit description in Basque
     *
     * @param descriptionBasque - Ambit description in Basque
     * */
    public void setDescriptionBasque(String descriptionBasque) {
        this.descriptionBasque = descriptionBasque;
    }

    /**
     * Method that obtains the subSubAmbit description in Catalan
     *
     * @return Ambit description in Catalan
     * */
    public String getDescriptionCatalan() {
        return descriptionCatalan;
    }

    /**
     * Method that sets the new subSubAmbit description in Catalan
     *
     * @param descriptionCatalan - Ambit description in Catalan
     * */
    public void setDescriptionCatalan(String descriptionCatalan) {
        this.descriptionCatalan = descriptionCatalan;
    }

    /**
     * Method that obtains the subSubAmbit description in Dutch
     *
     * @return Ambit description in Dutch
     * */
    public String getDescriptionDutch() {
        return descriptionDutch;
    }

    /**
     * Method that sets the new subSubAmbit description in Dutch
     *
     * @param descriptionDutch - Ambit description in Dutch
     * */
    public void setDescriptionDutch(String descriptionDutch) {
        this.descriptionDutch = descriptionDutch;
    }

    /**
     * Method that obtains the subSubAmbit description in Galician
     *
     * @return Ambit description in Galician
     * */
    public String getDescriptionGalician() {
        return descriptionGalician;
    }

    /**
     * Method that sets the new subSubAmbit description in Galician
     *
     * @param descriptionGalician - Ambit description in Galician
     * */
    public void setDescriptionGalician(String descriptionGalician) {
        this.descriptionGalician = descriptionGalician;
    }

    /**
     * Method that obtains the subSubAmbit description in German
     *
     * @return Ambit description in German
     * */
    public String getDescriptionGerman() {
        return descriptionGerman;
    }

    /**
     * Method that sets the new subSubAmbit description in German
     *
     * @param descriptionGerman - Ambit description in German
     * */
    public void setDescriptionGerman(String descriptionGerman) {
        this.descriptionGerman = descriptionGerman;
    }

    /**
     * Method that obtains the subSubAmbit description in Italian
     *
     * @return Ambit description in Italian
     * */
    public String getDescriptionItalian() {
        return descriptionItalian;
    }

    /**
     * Method that sets the new subSubAmbit description in Italian
     *
     * @param descriptionItalian - Ambit description in Italian
     * */
    public void setDescriptionItalian(String descriptionItalian) {
        this.descriptionItalian = descriptionItalian;
    }

    /**
     * Method that obtains the subSubAmbit description in Portuguese
     *
     * @return Ambit description in Portuguese
     * */
    public String getDescriptionPortuguese() {
        return descriptionPortuguese;
    }

    /**
     * Method that sets the new subSubAmbit description in Portuguese
     *
     * @param descriptionPortuguese - Ambit description in Portuguese
     * */
    public void setDescriptionPortuguese(String descriptionPortuguese) {
        this.descriptionPortuguese = descriptionPortuguese;
    }
}
